docs/src/manuals/inference/nonconjugate.md

```diff
@@ -0,0 +1,131 @@
+
+## Non-conjugate Structures
+
+However, models often contain non-conjugate structures, which prevent `RxInfer` from performing efficient inference. Non-conjugate priors occur when the prior and the likelihood do not result in a posterior that belongs to the same family as the prior. This complicates the inference process because it requires approximations or numerical methods instead of simple analytical updates.
+
+### Example Scenario
+
+Consider the scenario where we assign the `Beta` distribution as a prior for the mean parameter of a `Normal` distribution. Let's explore what happens in this case with an example.
+
+First, we generate some synthetic data:
+
+```@example non-conjugacy-prior-likelihood
+using Distributions, ExponentialFamily, Plots, StableRNGs
+
+# The model will infer the hidden parameters from data
+hidden_mean         = 0.2
+hidden_precision    = 0.8
+hidden_distribution = NormalMeanPrecision(hidden_mean, hidden_precision)
+
+number_of_datapoints = 1000
+data = rand(StableRNG(42), hidden_distribution, number_of_datapoints)
+
+histogram(data; normalize = :pdf)
+```
+
+Next, we specify the model. Suppose we believe the data follows a `Normal` distribution, and we are confident that the mean parameter is between `0` and `1`. The `Beta` distribution is a logical choice for the prior of the mean parameter because it models a continuous variable in the range from `0` to `1`. Similarly, we assign a `Beta` prior for the precision parameter, assuming it also lies between `0` and `1`.
+
+```@example non-conjugacy-prior-likelihood
+using RxInfer
+
+@model function non_conjugate_model(y)
+   m ~ Beta(1, 1)
+   p ~ Beta(1, 1)
+   y .~ Normal(mean = m, precision = p)
+end
+```
+
+If we attempt inference with this model, `RxInfer` will throw an error because the necessary computational rules for such a model are not available in closed form. This is due to the non-conjugate nature of the priors used.
+
+## Addressing Non-conjugacy with ExponentialFamilyProjection
+
+To overcome this limitation, `RxInfer` integrates with the `ExponentialFamilyProjection` package. This package re-projects non-conjugate relationships back into a member of the exponential family at the cost of some accuracy.
+
+!!! note
+    `RxInfer` supports non-conjugate inference for completeness, but be aware that inference execution times may increase significantly. This is because non-conjugate models require more complex computations, often involving sampling-based approximations.
+
+### Specifying Constraints
+
+The projection constraint must be specified using the `@constraints` macro. For example:
+
+```@example non-conjugacy-prior-likelihood
```

Comment:
Should `dist`  from ` ProjectedTo(dist)` be of the same form as factors of the model specification? For example, can I say `q(m)::ProjectedTo(Gamma)` given the model from the example?